### DISCUSSION TITLE
Winget or Scoop? Which one to choose when there are duplicate packages?

### DISCUSSION THREAD

**Discussion by venkat2305** [Devolutions/UniGetUI#309] [Q&A]
what should i prefer , winget or scoop. if a program is available in both, which now should i prefer , what are the advantages of these package managers. Im new to this. what's chocolatey package manager.
tldr; git is present in both winget and scoop. which one should  i use to install git?

**Comment by marticliment** [comment]
Well, it depends.


## Installing apps:
Winget is a package manager that will download the program installer and install it as you would do manually in your system. The final effect will be an app installed in a _standard_ way. Those apps could be uninstalled from, let's say, the Windows Settings op the Control Panel. Basically, winget downloads the official installers and runs it with some sort of `--silent --norestart --unattended` flags.

Scoop, however, aims to remove installers and downloads the executables, places them on a custom directory and manually creates any shortcuts if required. To uninstall an app, youu would require scoop to uninstall the app, bevause the installation is not registered in the system. 

Additionally, the scoop package manager will easily allow you to properly install and older version of a scoop app (you might succeed with winget, but things can really get messed up).
However, some apps that have been repackaged to be installed with scoop might fail to install some features/might require manual run of post-installation scripts, since the lack of the installer on some packages.


## Updating apps

Both package managers will be able to update their own apps. (In fact, winget will detect if it can update any app present in your system, and not only the opes installed through the package manager.). Some specific packages might cause issues with winget upgrade, but WingetUI will esaily allow you to blacklist a specific package to avoid issues with it.

## Uninstalling apps
Each package manager takes care of their apps, so no issues should happen there. 
It should be noted also that winget provides the ability to uninstall any app installed in the system, whether it is published on winget or not.

**Reply by venkat2305** [reply]
Scoop seems better to me overall. And what about chocolatey?

**Reply by ppvnf** [reply]
> Scoop seems better to me overall. And what about chocolatey?

Chocolatey is shareware (some features such as `choco sync` aren't available for free). I don't know exactly how it works, but I believe Chocolatey also prefers regular installation like Winget (it also offers portable installs sometimes). Another feature it shares with Winget is av scan (I am unsure if this is done for Scoop packages). It seems like Chocolatey packages are auto updated, but sometimes the package maintainer disappears and if this happens there's not really much you or someone else can do in case a problem arises, since only the maintainer is able to fix the package (on the other hand, anyone can submit PRs to fix Scoop and Winget broken packages)

**Reply by ppvnf** [reply]
Chocolatey has some neat features tho, like being able to easily use install parameters, such as removing desktop icon, installing additional components etc.

**Reply by marticliment** [reply]
Chocolatey uses a mixture, and can handle both fully portable, scoop like apps and more conventional winget installers

**Reply by eugenesvk** [reply]
> To uninstall an app, youu would require scoop to uninstall the app, bevause

Or you could just delete the folder with the app

**Comment by ppvnf** [comment]
As [martinet101](https://github.com/martinet101) has explained, Scoop and Winget work very differently from each other, even though they achieve similar things. @venkat2305, if you see Git in the info menu you can see in the field "Installer URL" that Scoop "installs" the portable version while Winget uses the .exe official installer. Both approaches have pros and coins as he said. 

I don't think there's one package manager that is better than the other, but I have a tendency to prefer Winget packages, since a few apps may not work properly with portable version. However, if I want to test an app I usually go with Scoop as official installers sometimes do not uninstall cleanly. 

Besides, IMO a big advantage to Scoop is the native auto-update feature. Winget doesn't natively support this, however a few people such as https://github.com/vedantmgoyal2009/vedantmgoyal2009 are running scripts that automatically submit new app versions to Winget, which in turn makes package updates in Winget slower when compared to Scoop.

**Comment by venkat2305** [comment]
@ppvnf @martinet101 for someone using widget UI, if a program is offered in multiple package managers, what will be your advice on choosing the package manager? 
which type of programs should be preferred in scoop(examples?) and which should be preferred in Winget(examples?)

**Reply by marticliment** [reply]
Any app that you can found portable in scoop you can install it safely and it won't give any issue. But if the apps are nonportable (from buckets like the extras or the nonportable ones), i highly recommend winget. 
However, if you need apps to install without adiminstrator rights you might also go scoop.

**Reply by marticliment** [reply]
There's also another thing to take in account, and it is that while in winget you'll find apps in the classic sense of desktop, graphical application, on scoop you'll be able to find cli utilities, toolkits and sdks that you couldn't download from winget at all.

**Reply by ppvnf** [reply]
Most of the time I choose Winget for the reason mentioned by martinet101, but sometimes I use Scoop for reason such as: 

- Winget still lacks support for .zip 
- Apps are usually updated faster on Scoop
- Occasionally I prefer the portable version of certain apps, specially if I plan to remove them from my PC shortly, because installers (standalone or downloaded by Winget) sometimes don't remove all program files

**Comment by venkat2305** [comment]
1. if the same program has two different package IDs for scoop and Winget package managers, we can install both of them at the same time as two different programs, correct?
2. node js is available in both winget and scoop, which package manager should I prefer?

**Reply by marticliment** [reply]
> * if the same program has two different package IDs for scoop and Winget package managers, we can install both of them at the same time as two different programs, correct?

There shouldn't be any issues on that



> 2\. node js is available in both winget and scoop, which package manager should I prefer?

i have it through winget and it works just fine. Also, apps that can be auto-updare through themselves probably is better to install them through winget

**Comment by rumplin** [comment]
Personally, I disable chocolate the first thing I do after installing WingetUI. I relly completly on winget as source of apps, so far I got everything I needed from winget.
